pkg/controller/statefulset/stateful_set.go

```diff
@@ -309,11 +335,20 @@ func (ssc *StatefulSetController) deletePod(logger klog.Logger, obj interface{})
 // NOTE: Returned Pods are pointers to objects from the cache.
 // If you need to modify one, you need to copy it first.
 func (ssc *StatefulSetController) getPodsForStatefulSet(ctx context.Context, set *apps.StatefulSet, selector labels.Selector) ([]*v1.Pod, error) {
-	// List all pods to include the pods that don't match the selector anymore but
-	// has a ControllerRef pointing to this StatefulSet.
-	pods, err := ssc.podLister.Pods(set.Namespace).List(labels.Everything())
-	if err != nil {
-		return nil, err
+	// List all pods to include the pods that has a ControllerRef pointing to this StatefulSet or Orphan pods with match labels.
```

Comment:
comment is talking about labels, where are we using/comparing labels?

Reply:
Basically orphan pods labels are used in [claimPods](https://github.com/kubernetes/kubernetes/pull/130806/files#diff-3444629cff8d63a4b7c9f59da33ac22d6ce78b7361ae6995b84592cb24c29b45R367) to decide if they can be adopted.
Added a bit more detailed comment.

Reply:
gotcha. thanks!